[2025-07-15T16:18:27.814+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flight_etl_dag.loading_data manual__2025-07-15T16:18:15.142717+00:00 [queued]>
[2025-07-15T16:18:27.824+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flight_etl_dag.loading_data manual__2025-07-15T16:18:15.142717+00:00 [queued]>
[2025-07-15T16:18:27.826+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-07-15T16:18:27.846+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): loading_data> on 2025-07-15 16:18:15.142717+00:00
[2025-07-15T16:18:27.854+0000] {standard_task_runner.py:60} INFO - Started process 111 to run task
[2025-07-15T16:18:27.859+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'flight_etl_dag', 'loading_data', 'manual__2025-07-15T16:18:15.142717+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/flight_etl_dag.py', '--cfg-path', '/tmp/tmp054xsce7']
[2025-07-15T16:18:27.860+0000] {standard_task_runner.py:88} INFO - Job 8: Subtask loading_data
[2025-07-15T16:18:27.955+0000] {task_command.py:423} INFO - Running <TaskInstance: flight_etl_dag.loading_data manual__2025-07-15T16:18:15.142717+00:00 [running]> on host 3eb6f9d291a3
[2025-07-15T16:18:28.074+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='flight_etl_dag' AIRFLOW_CTX_TASK_ID='loading_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-15T16:18:15.142717+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-15T16:18:15.142717+00:00'
[2025-07-15T16:18:28.291+0000] {logging_mixin.py:188} INFO - Data to load:
[2025-07-15T16:18:28.305+0000] {logging_mixin.py:188} INFO -   aircraft_id flight_number origin_country  ...  position_source      time       date
0      c07d0b      CGVJE            Canada  ...            ADS-B  11:17:55 2025-07-15
1      a5f852      RTY484    United States  ...            ADS-B  11:17:55 2025-07-15
2      ab1644      UAL2364   United States  ...            ADS-B  11:17:55 2025-07-15
3      ac8fbd      N9083Z    United States  ...            ADS-B  11:17:56 2025-07-15
4      aa3cbe      N759PA    United States  ...            ADS-B  11:17:23 2025-07-15

[5 rows x 11 columns]
[2025-07-15T16:18:28.368+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/load.py", line 20, in load_to_db
    create_tables()
  File "/opt/airflow/db_schema.py", line 8, in create_tables
    cursor.execute("""
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "pg_type_typname_nsp_index"
DETAIL:  Key (typname, typnamespace)=(aircraft, 2200) already exists.

[2025-07-15T16:18:28.409+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=flight_etl_dag, task_id=loading_data, execution_date=20250715T161815, start_date=20250715T161827, end_date=20250715T161828
[2025-07-15T16:18:28.436+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 8 for task loading_data (duplicate key value violates unique constraint "pg_type_typname_nsp_index"
DETAIL:  Key (typname, typnamespace)=(aircraft, 2200) already exists.
; 111)
[2025-07-15T16:18:28.475+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-07-15T16:18:28.503+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
